src/components/CheckoutForm.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState } from 'react';
+import { Col, Row, Form } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
+import { useHistory } from "react-router-dom";
+import { AppDispatch } from '../actions';
+import { checkoutCart } from '../actions/cart';
+import { PaymentInfo, ContactInfo, Address, CheckoutData } from '../types/checkout';
+
+const CheckoutForm = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const history = useHistory();
+  const [firstName, setFirstName] = useState();
+  const [lastName, setEmail] = useState();
+  const [email, setLastName] = useState();
+  const [street1, setStreet1] = useState();
+  const [street2, setStreet2] = useState();
+  const [country, setCountry] = useState();
+  const [state, setState] = useState();
+  const [zip, setZip] = useState();
+
+  const [ccName, setCcName] = useState();
+  const [ccNumber, setCcNumber] = useState();
+  const [ccExpiration, setCcExpiration] = useState();
+  const [ccCVV, setCcCVV] = useState();
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const address: Address = {
+      street1,
+      street2,
+      country,
+      state,
+      zip,
+    };
+    
+    const contactInfo: ContactInfo = {
+      firstName,
+      lastName,
+      email,
+    };
+
+    const paymentInfo: PaymentInfo = {
+      ccName,
+      ccNumber,
+      ccExpiration,
+      ccCVV,
+      shippingAddress: address,
+      billingAddress: address,
+    };
+
+    const checkoutData: CheckoutData = {
+      contactInfo,
+      paymentInfo,
+      products: [],
+    };
+
+    dispatch(checkoutCart(history, checkoutData));
```

Comment:
@timchin passing `history` into the dispatch to navigate to the Thank You page on a successful checkout.